elasticdl/python/common/k8s_volume.py

```diff
@@ -16,32 +22,54 @@ def parse_volume_and_mount(volume_conf, pod_name):
         volumes (List): a Python list contains k8s volumes.
         volume_mounts (List): a Python list contains k8s volume mounts.
     """
-    volumes = []
+    volumes_cache = {}
+    volumes_cache["pvc"] = {}
+    volumes_cache["host_path"] = {}
     volume_mounts = []
     volume_dicts = parse(volume_conf)
-    for i, volume_dict in enumerate(volume_dicts):
-        volume_name = pod_name + "-volume-%d" % i
+    num_volumes = 0
+    for volume_dict in volume_dicts:
         if "claim_name" in volume_dict:
-            pvc_volume_source = client.V1PersistentVolumeClaimVolumeSource(
-                claim_name=volume_dict["claim_name"], read_only=False
-            )
-            volume = client.V1Volume(
-                name=volume_name, persistent_volume_claim=pvc_volume_source
-            )
+            volume = volumes_cache["pvc"].get(volume_dict["claim_name"], None)
```

Comment:
An example may be needed to explain when the volume isn't None?

Reply:
We can check the example in the PR description: 
```
--volume='claim_name=glusterfs-elasticdl,mount_path=/assets,sub_path=elasticdl/20200312/vocab;claim_name=glusterfs-elasticdl,mount_path=/work_dir' \
```

Reply:
Will add this example into the comments.